### PR TITLE
Remove try-except block from fixture `fanouthosts`

### PR DESCRIPTION
**Pull request by lizhijianrd**
### Description of PR

Summary:
The `try-except` block in fixture `fanouthosts` simply ignore the exceptions caught. This may hide wired issues and make debugging more difficult. We'd better let the issue exposed and be fixed. So, in this PR, I removed the `try-except` block.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
The `try-except` block in fixture `fanouthosts` simply ignore the exceptions caught. This may hide wired issues and make debugging more difficult. We'd better let the issue exposed and be fixed. So, in this PR, I removed the `try-except` block.

#### How did you do it?
1. Removed the `try-except` block.
2. Skip processing for KVM DUTs, because they don't have fanout.

#### How did you verify/test it?
Verified on virtual testbed.

### CODE REVIEW

**Comment by lizhijianrd** [comment]
Hi @wangxin @Blueve, do we need to cherry-pick this PR to 202205/202012 branch?

File: tests/conftest.py [hunk 1]
```diff
@@ -529,71 +529,71 @@ def fanouthosts(ansible_adhoc, conn_graph_facts, creds, duthosts):      # noqa F
     dev_conn = conn_graph_facts.get('device_conn', {})
     fanout_hosts = {}
     # WA for virtual testbed which has no fanout
```
[the remark this answers is not in the document]
Comment (wangxin) [reply tests/conftest.py:531]: `WA` means workaround.

[the remark this answers is not in the document]
Comment (lizhijianrd) [reply tests/conftest.py:531]: @wangxin Thanks. It can pass the PR test, is that enough?

**Review by wangxin — APPROVED** [review]

**Comment by lizhijianrd** [comment]
> Hi @wangxin @Blueve, do we need to cherry-pick this PR to 202205/202012 branch?

As discussed offline, we will run some test pipeline to verify this PR. If it's not harmful on master, then we will cherry-pick to 202205 and 202012.

**Pull request merged**